spyder_kernels/console/kernel.py

```diff
@@ -660,10 +662,39 @@ def set_color_scheme(self, color_scheme):
         if color_scheme == "dark":
             # Needed to change the colors of tracebacks
             self.shell.run_line_magic("colors", "linux")
-            self.set_sympy_forecolor(background_color='dark')
         elif color_scheme == "light":
             self.shell.run_line_magic("colors", "lightbg")
-            self.set_sympy_forecolor(background_color='light')
+        self.set_sympy_forecolor(background_color=color_scheme)
+        self.set_traceback_highlighting(color_scheme)
+
+    def set_traceback_highlighting(self, color_scheme):
+        """Set the traceback highlighting color."""
+        color = 'bg:ansired' if color_scheme == 'dark' else 'bg:ansiyellow'
+        from IPython.core.ultratb import VerboseTB
+
+        if getattr(VerboseTB, 'tb_highlight', None) is not None:
+            VerboseTB.tb_highlight = color
+        elif getattr(VerboseTB, '_tb_highlight', None) is not None:
+            VerboseTB._tb_highlight = color
+
+    def set_traceback_syntax_highlighting(self, syntax_style):
+        """Set the traceback syntax highlighting style."""
+        try:
+            import IPython.core.ultratb
+            from IPython.core.ultratb import VerboseTB
+
+            from spyder.plugins.ipythonconsole.utils.style import create_style_class
```

Comment:
> Should the logic related with the creation of a Pygments style class while Spyder is running be moved or copied to spyder-kernels @ccordoba12 ?

To be able to support this in a Spyder bugfix release (i.e. 6.0.3), that code would need to be copied to Spyder-kernels (otherwise we'd need to require people to update spyder-kernels to 3.0.2 in all their envs). Then, in 6.1 we could remove that code from Spyder and import it directly from spyder-kernels.

> Maybe a package with those utility functions should be created?

I don't think that's necessary, i.e. having that code in spyder-kernels should be enough because Spyder depends directly on spyder-kernels.

Reply:
But I think you'd need to send to the kernel the result of the `get_color_scheme` function (in `spyder.config.gui`). In other words, that function shouldn't be moved and it shouldn't be hard to send its result to the kernel because it's probably a dict.